### PR TITLE
Allow custom error transformation on GraphQLEngine #119 #120

### LINKED ISSUE #119: Move error transformation to dedicated class
Hi Tommy, we're playing with the idea to move the custom error transformations (see https://github.com/graphql-dotnet/conventions/blob/master/src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs#L288) to a separate class, which can be easily replaced by a new method called `WithErrorTransformation` or `WithErrorTransformer` f.e.

What would you think about that?

### PR DESCRIPTION
**Pull request by huysentruitw**

### CODE REVIEW

File: src/GraphQL.Conventions/Adapters/Engine/ErrorTransformations/IErrorTransformation.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace GraphQL.Conventions.Adapters.Engine.ErrorTransformations
+{
+    public interface IErrorTransformation
+    {
+        Task<ExecutionErrors> Transform(ExecutionErrors errors);
```
Comment (tlil) [inline src/GraphQL.Conventions/Adapters/Engine/ErrorTransformations/IErrorTransformation.cs]: Any particular reason why you want to return this as a `Task<>`? Any cases where the error transformation would be asynchronous?

**Comment by tlil** [comment]
Other than that, this looks good to me. Thanks!

File: src/GraphQL.Conventions/Adapters/Engine/ErrorTransformations/IErrorTransformation.cs (hunk 1 above)
Comment (huysentruitw) [reply src/GraphQL.Conventions/Adapters/Engine/ErrorTransformations/IErrorTransformation.cs]: I was doubting between the two options, you're right will change it to non-async.

**Review by tlil — APPROVED** [review]

**Comment by tlil** [comment]
Thanks for this

**Pull request merged**